### ISSUE TITLE
GSE111976 -- human endometrium during menstrual cycle

### ISSUE THREAD

**Issue by rays22** [ebi-ait/hca-ebi-wrangler-central#139]
**Primary Wrangler:**
Ray
**Secondary Wrangler:**
Marion

**Associated files:**

Google Drive:
[GSE111976-HUTER-endometrium_MC](https://drive.google.com/drive/folders/1kxtAeVMp7HDJ6HyGFsQ3PRRra2R81wuf)

**Published study links**

Paper:
Wang W, Vilella F, Alama P, et al. Single-cell transcriptomic atlas of the human endometrium during the menstrual cycle. Nature Medicine. 2020 Sep. [DOI: 10.1038/s41591-020-1040-z](https://doi.org/10.1038/s41591-020-1040-z).  PMID:32929266

Accessioned data:
[GSE111976](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE111976)

**Key Events**

- [x] convert published metadata to HCA spreadsheet
- [x] manually curate dataset to meet HCA metadata standard
- [x] Upload sheet to validate metadata
- [x] Check linking using ingest graph validator
- [x] Transfer files to ingest to validate data files
- [x] Collect cell-gene expression matrices and cell type annotations if available
- [x] Ask the Secondary Wrangler for an end-to-end review of the project. Ask the Expertise Wrangler to review specific tabs if needed
- [x] Submit dataset to Production

**Comment by rays22** [comment]
This dataset is part of the HUTER project. See <https://github.com/ebi-ait/hca-ebi-wrangler-central/issues/138>

**Comment by rays22** [comment]
* [ ] Use this dataset to review the HUTER clinical information and metadata variables. Identify any gaps or incompatibilities in HUTER metadata by comparison with HCA standards.
 * [ ] Check the metadata in this dataset and if appropriate, provide the HUTER team a general summary of the ontologies we expect them to use so they can build that info into their own platform.

**Comment by rays22** [comment]
### Gene by cell matrices and cell type annotations

I have uploaded the GEO Supplementary files

```
GSE111976_ct.csv.gz: gene by cell table for C1 dataset.
GSE111976_ct_endo_10x.rds.gz: gene by cell table for 10x dataset.
GSE111976_summary_C1_day_donor_ctype.csv: for each analyzed cell in the C1 dataset, annotation on the donor, day of the menstrual cycle, and cell type.
GSE111976_summary_10x_day_donor_ctype.csv: for each analyzed cell in the 10x dataset, annotation on the donor, day of the menstrual cycle, and cell type.
```

to the [gdrive folder](https://drive.google.com/drive/folders/1aJitoxjUF10_6rVAnojkzyZgy74NGHdZ) of the dataset.

**Comment by rays22** [comment]
* I have transferred the original fastq data files (`GSE111976-endometrium_MC-issue139`) to the wragler/contributor upload area.
* I have filled in and uploaded the HCA metadata spreadsheet
to the [gdrive folder](https://drive.google.com/drive/folders/1kxtAeVMp7HDJ6HyGFsQ3PRRra2R81wuf) of the dataset.
    * The metadata spreadsheet is ready for secondary review.

**Comment by mshadbolt** [comment]
Hi @rays22 I will take on secondary review of this over the next couple of days.

**Comment by mshadbolt** [comment]
Hi @rays22 
I have just started looking at the spreadsheet and I am wondering whether you have asked the author of the paper whether they can provide more specific metadata about each donor? 
e.g. Age, Ethnicity, BMI

**Comment by rays22** [comment]
> Hi @rays22
> I have just started looking at the spreadsheet and I am wondering whether you have asked the author of the paper whether they can provide more specific metadata about each donor?
> e.g. Age, Ethnicity, BMI

No, and I do not think it would be a good idea to obtain unpublished sensitive metadata about each donor.

In the `Population characteristics` section of the supplement they provide aggregate information on 'de-identified' living donors, such as age and BMI. Apparently, they chose not to disclose that information for individual donors. I trust their judgement.
Even if they were to reveal to the HCA some new, genetically identifiable, individual human donor data, would that not be a problem for the HCA from a legal data protection perspective?

**Comment by mshadbolt** [comment]
I don't think that a 5 year age range, ethnicity and BMI would be considered sensitive information or allow a 'healthy living female' to be identified, genetically or otherwise. I thought that given we have a relationship with the author it would be worth asking whether this information was intentionally left out due to ethics or consent requirements or was simply not included. 

I have had examples where asking in these instances has yielded better metadata from the contributor. I see it as valuable to ask given we are trying to curate to a higher standard than what is currently in the public domain so that the data becomes more useful to consumers.

I don't know if it would be a problem from a legal/data protection perspective in this instance. @lauraclarke if contributors provide us directly with more metadata about living donors than is published in the archive, could that mean we can't pass it on to DCP due to GDPR?

**Comment by rays22** [comment]
> I thought that given we have a relationship with the author
FYI: We do not have a relationship with the authors of this paper. Some of the authors of this paper are members of the HUTER project (but this paper is not part of HUTER), but they are not the people we have had meetings with.
This is not a direct contribution from a lab. The data and metadata is coming from public GEO data and the paper.

**Comment by mshadbolt** [comment]
hi @rays22 I found some issues with missing files/cell suspensions that might have happened during the conversion from geo:
GSM4577312 cell suspension id is in the cell suspension tab but not in the sequence file tab

Following cell suspensions that are listed in the sequence file tab are not in the cell suspension tab tab:

|cell_suspension.biomaterial_core.biomaterial_id |  n|
|:-----------------------------------------------|--:|
|GSM3046269                                      |  2|
|GSM3046270                                      |  2|
|GSM3046271                                      |  2|
|GSM3046272                                      |  2|
|GSM3046273                                      |  2|
|GSM3046274                                      |  2|
|GSM3046275                                      |  2|
|GSM3046276                                      |  2|
|GSM3046277                                      |  2|
|GSM3046278                                      |  2|
|GSM3046279                                      |  2|
|GSM3046280                                      |  2|
|GSM3046281                                      |  2|
|GSM3046282                                      |  2|
|GSM3046283                                      |  2|
|GSM3046284                                      |  2|
|GSM3046285                                      |  2|
|GSM3046286                                      |  2|
|GSM3046287                                      |  2|
|GSM3046288                                      |  2|
|GSM3046289                                      |  2|
|GSM3046290                                      |  2|
|GSM3046291                                      |  2|
|GSM3046292                                      |  2|
|GSM3046293                                      |  2|
|GSM3046294                                      |  2|
|GSM3046295                                      |  2|
|GSM3046296                                      |  2|
|GSM3046297                                      |  2|
|GSM3046298                                      |  2|
|GSM3046299                                      |  2|
|GSM3046300                                      |  2|
|GSM3046301                                      |  2|
|GSM3046302                                      |  2|
|GSM3046303                                      |  2|
|GSM3046304                                      |  2|
|GSM3046305                                      |  2|
|GSM3046306                                      |  2|
|GSM3046307                                      |  2|
|GSM3046308                                      |  2|
|GSM3046309                                      |  2|
|GSM3046310                                      |  2|
|GSM3046311                                      |  2|
|GSM3046312                                      |  2|
|GSM3046313                                      |  2|
|GSM3046314                                      |  2|
|GSM3046315                                      |  2|
|GSM3046316                                      |  2|
|GSM3046317                                      |  2|
|GSM3046318                                      |  2|
|GSM3046319                                      |  2|
|GSM3046320                                      |  2|
|GSM3046321                                      |  2|
|GSM3046322                                      |  2|
|GSM3046323                                      |  2|
|GSM3046324                                      |  2|
|GSM3046325                                      |  2|
|GSM3046326                                      |  2|
|GSM3046327                                      |  2|
|GSM3046328                                      |  2|
|GSM3046329                                      |  2|
|GSM3046330                                      |  2|
|GSM3046331                                      |  2|
|GSM3046332                                      |  2|
|GSM3046333                                      |  2|
|GSM3046334                                      |  2|
|GSM3046335                                      |  2|

**Comment by rays22** [comment]
Thanks @mshadbolt . Good catch.
I did some investigation. I have found that the missing cell suspensions all came from a donation at menstrual cycle 7. This narrows down the potential donors to `D04` and `D05`.

* The "day" field in supplementary table `GSE111976_summary_C1_day_donor_ctype.csv.gz` can be matched with the "!Sample_characteristics_ch1 time in menstrual cycle: day" field in `GSE111976-GPL18573_series_matrix.txt.gz`.
In turn, the `cell_name` field in`GSE111976_summary_C1_day_donor_ctype.csv.gz` can be matched with the "!Sample_title" field in `GSE111976-GPL18573_series_matrix.txt.gz`.
* Based on the "day" : "7" value of menstrual cycle and the "cell_name" values in supplementary table `GSE111976_summary_C1_day_donor_ctype.csv.gz`, the missing cell suspensions could have come from donors `D04` and/or `D05`.

Actions:
- [ ] Before I can add the missing cell suspensions, I need to query the authors of the paper about the correct donor information of the 67 cell suspensions that are listed in the `Sequence file` tab, but are missing from the `Cell suspension` tab of the metadata spreadsheet.
- [ ] At the same time, I can ask them if they could share the individual age, ethnicity and BMI donor values (maybe as managed access?).

**Comment by rays22** [comment]
I have added the missing 67 cell suspensions (all from donor D05) based on information from author Wanxin Wang, and uploaded an [updated metadata spreadsheet to the gdrive.](https://drive.google.com/file/d/1WiI8_4RLOWJQ34IpAa1VfbAhkrh8Sh0e/view?usp=sharing)

**Comment by rays22** [comment]
> I have just started looking at the spreadsheet and I am wondering whether you have asked the author of the paper whether they can provide more specific metadata about each donor?
> e.g. Age, Ethnicity, BMI

@mshadbolt : I am afraid there are privacy policy issues that prevent the sharing of these individual donor characteristics.
One of the authors says in the response to my query:
> Do our restrictions in our IRB, all the patients were in a specific range of age, we do not have the concrete age per every patient, the range was between 18 to 34. In the subject details in the Methods section you have the info.

**Comment by lauraclarke** [comment]
@rays22 can you put in the age bin?

**Comment by rays22** [comment]
> @rays22 can you put in the age bin?

It is in the free text of `donor_organism.biomaterial_core.biomaterial_description`:

>The donor is one of a group of healthy and fertile women aged 18-34 years with regular menstrual cycling (3-4 d every 28-30 d), body mass index 19-29 kg m-2 (inclusive), normal karyotype and negative serological tests for human immunodeficiency virus, hepatitis B virus, hepatitis C virus and syphilis. Women with the following conditions were excluded from tissue collection: recent contraception (intrauterine device usage in past 3 months; hormonal contraceptives in past 2 months), uterine pathology (endometriosis, leiomyoma or adenomyosis; bacterial, fungal or viral infection) or polycystic ovary syndrome.

**Comment by lauraclarke** [comment]
But I thought you could put binned ages in the age field, is there a reason not to add it there too, then it will be accessible to the browser

**Comment by rays22** [comment]
> But I thought you could put binned ages in the age field, is there a reason not to add it there too, then it will be accessible to the browser

Oh, I see.
OK, I'll add the binned ages.

**Comment by rays22** [comment]
> OK, I'll add the binned ages.

I have updated the [metadata spreadsheet](https://drive.google.com/file/d/1WiI8_4RLOWJQ34IpAa1VfbAhkrh8Sh0e/view?usp=sharing).

**Comment by rays22** [comment]
- [ ] Upload dataset to Ingest UI
- [ ] Convert into SCEA format
- [ ] Export for SCEA

**Comment by rays22** [comment]
I am blocked here:
I have transferred 4347 data files using `hca-util sync` from the wrangler/contributor upload area to the Ingest production s3 bucket.
I have checked and confirmed that all the files have been transferred and they are all in the destination Ingest upload area.
However, Ingest UI reports that `Data files: 1867 files missing`. 
I have tried re-uploading the files using `hca-util` again, but that did not solve the issue.
I need developer help to unblock the missing file issue in Ingest.

**Comment by rays22** [comment]
> I am blocked here:
> I have transferred 4347 data files using `hca-util sync` from the wrangler/contributor upload area to the Ingest production s3 bucket.
> I have checked and confirmed that all the files have been transferred and they are all in the destination Ingest upload area.
> However, Ingest UI reports that `Data files: 1867 files missing`.
> I have tried re-uploading the files using `hca-util` again, but that did not solve the issue.
> I need developer help to unblock the missing file issue in Ingest.

This issue may be related: <https://github.com/ebi-ait/dcp-ingest-central/issues/159> .

**Comment by yusra-haider** [comment]
this issue is because of this issue: https://github.com/ebi-ait/hca-ebi-dev-team/issues/370#issuecomment-760179407


here's what I've tried so far:

- for some reason, the `sync` script for `hca-util` fails to assign the `ContentType` as `dcp-type=data` for some files in this dataset. the print statements show that the ContentType does get set to `dcp-type=data`  but it doesn't actually get reflected after the files are copied. This is only happening for a specific subset of files in this dataset, and they seem to be the same files every time.

- the issue reproduced multiple times on staging, and the same number of files is invalid every time

- I updated the `boto3` library to the latest version (boto3==1.17.4 & botocore==1.20.4) and tried syncing from Ray's upload area to the ingest staging upload area and it didn't make any difference

- I also downloaded one invalid file and one valid file (1772158101_C01_A3__N311Barcode_719-513_S291_R1_001.fastq.gz
and endo_14_S1_L001_I1_001.fastq.gz respectively), uploaded to a new upload area in `hca-util` and synced with ingest staging upload area, and that did the trick -- the previously invalid file got validated, and had the `ContentType` set as `dcp-type=data` 

- I don't know what's causing this to happen, but for now one possible solution could be to do a normal sync from Ray's upload area in hca-util to the ingest upload area, followed by uploading all invalid files to a another upload area in hca-util, using the `hca-util upload` command followed by syncing to ingest.


for reference, all files in  hca-util upload area, for this dataset were obtained from NCBI cold storage buckets and transferred directly via aws commands

I'm also attaching the file with invalid filenames for reference
[invalid.json](https://github.com/ebi-ait/hca-ebi-wrangler-central/files/5957219/invalid.json)

**Comment by rays22** [comment]
1. I have downloaded the fastq data files from the `hca-util` s3 bucket.
2. I have created a new `hca-util` upload area.
3. I have tried to upload the files in bulk to the new `hca-util` upload area in bulk (using command: `hca-util upload .`), but the upload failed after uploading about ~100 files in 10 hours. I tried that again with the same result.
4. I have successfully uploaded the remaining fastq files to the new `hca-util` upload area one-by-one (using `hca-util upload <filename>` commands).
5. I used `hca-util sync` to re-upload from the new `hca-util` upload area the fastq files that were falsely reported by Ingest to be missing from the Ingest upload area for the submission. This process was successful and triggered validation in Ingest. One file was found to be invalid by Ingest:

>: ERROR: Error in file /data/a489cab8-2e35-456f-8b28-3d571a1bf213/1782163192_C36_F9__N345Barcode_726-508_S129_R2_001.fastq.gz: line 4813284: file truncated

I have downloaded the original data file from the `hca-util` upload area for manual checks. I can confirm that the original data file is corrupted and not a valid gz compressed fastq file. I am omitting it and its paired read as well the corresponding cell suspension (GSM3047806, which is one Fludigm cell) from the dataset, because or their limited usefulness due the loss of sequence data integrity. I can update the dataset once the authors update the corrupted file in the NCBI archives.

6. All the 4345 sequence data files and the metadata in the set passed validation by Ingest.

**Comment by ami-day** [comment]
hca to scea tool not working for @rays22 . We have decided to install it on EC2 so that the same installtion/repo is available to all users. In the meantime, I have converted this dataset locally, and added them to the dataset folder (same as hca google sheet). It needs further manual curation before sending to SCEA team.

**Comment by rays22** [comment]
I have cleaned up redundant aws s3 file storage areas related to this project to free up approximately 1.4 TB space.

**Comment by rays22** [comment]
I have collected the gene by cell matrix and cell annotation [files associated with this project.](https://drive.google.com/drive/folders/1aJitoxjUF10_6rVAnojkzyZgy74NGHdZ)

1. `GSE111976_ct.csv.gz`: gene by cell table for C1 dataset.
2. `GSE111976_ct_endo_10x.rds.gz`: gene by cell table for 10x dataset.
3. `GSE111976_umap_endo_10x.csv.gz`: umap on all cells in the 10x dataset (shown in Extended Data Fig. 2A).
4. `GSE111976_summary_10x_day_donor_ctype.csv`: for each analyzed cell in the 10x dataset, annotation on the donor, day of the menstrual cycle, and cell type.
5. `GSE111976_summary_C1_day_donor_ctype.csv`: for each analyzed cell in the C1 dataset, annotation on the donor, day of the menstrual cycle, and cell type.

**Comment by rays22** [comment]
* I am going to exclude the raw sequence files from the DCP export, because of living European donors and GDPR concerns.
* I have included all the `2,148` Fluidigm C1 (and also the `9` 10x) cell suspensions in the metadata together with the matrix and cell type annotation files.
* I have uploaded the metadata spreadsheet that includes the matrix/analysis file metadata and excludes the raw sequence data files [to the gdrive.](https://drive.google.com/file/d/1WIiSe_In47NvHupoiPPq_Qzl5I1R-4ph/)
* I had to hand edited `analysis_file.file._core.file_source` in Ingest UI, because Ingest did not accept `GEO` as `file_source` from the metadata spreadsheet. I selected `GEO` from the drop-down menu in Ingest UI and that solved the validation errors:

```
Validation Report
* should NOT have additional properties at root of document


  "validationErrors" : [ {
    "errorType" : "METADATA_ERROR",
    "message" : "should NOT have additional properties",
    "userFriendlyMessage" : "should NOT have additional properties at root of document",
    "absoluteDataPath" : "root of document"
  } ],

```
